sktime/transformations/series/fourier.py

```diff
@@ -184,6 +220,8 @@ def _transform(self, X, y=None):
             X_transformed[f"sin_{sp}_{k}"] = np.sin(int_index * 2 * k * np.pi / sp)
             X_transformed[f"cos_{sp}_{k}"] = np.cos(int_index * 2 * k * np.pi / sp)
 
+        # Ensure transformed X has same index
+        X_transformed.index = X.index
```

Comment:
do a deepcopy(X.index) here for good measure

Reply:
we're testing for side effects, using `test_methods_have_no_side_effects`, so I think that is not needed?